srml/contract/src/lib.rs

```diff
@@ -14,41 +14,68 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality for the runtime to deploy and execute smart-contracts expressed in WebAssembly.
+//! To add this functionality, you need to add the contract module to `construct_runtime!` in the runtime module.
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module extends accounts with the ability to create smart-contracts and make calls to the contracts.
+//! Each smart-contract is associated with an account, which handles the code and storage. When a smart-contract is called, its associated code gets executed.
+//! This code can alter the storage entries of the associated account, create new smart-contracts, or call other smart-contracts.
+//! 
+//! Senders must specify a gas limit with every call, as all instructions invoked by the smart-contract require gas.
+//! Unused gas is refunded after the call, regardless of the execution outcome. 
+//! 
+//! **NOTE:** If the gas limit is reached, then all calls and state changes (including balance transfers) are only reverted at the current call's contract level.
+//! For example, if contract A calls B and B runs out of gas mid-call, then all of B's calls are reverted.
+//! Assuming correct error handling by contract A, A's other calls and state changes still persist. 
+//! 
+//! **NOTE:** Call failures are also not always cascading. For example, if contract A calls contract B and B
+//! fails, A can decide how to handle that failure, either proceeding or reverting A's changes.
+//! 
+//! Finally, when the `balances` module determines an account is dead (i.e. account balance fell below the
+//! existential deposit), it reaps the account. This will delete the associated code and storage of the account.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable functions
+//! 
+//! * `put_code` - Stores the given binary Wasm code into the chains storage and returns its `codehash`.
+//! 
+//! * `create` - Deploys a new contract from the given `codehash`, optionally transferring some balance. This creates a new smart contract account and calls its contract deploy handler to initialize the contract.
+//! 
+//! * `call` - Makes a call to an account, optionally transferring some balance.
```

Comment:
That's a really good question! Surprisingly you can, sort of.
`call` and `instantiate` does a transfer as part of their functionality. In order to transfer some funds to a non-contract account, a contract should invoke the `ext_call` function with the account's address. But in this case, of course, no code will be executed.

Reply:
So if I understand, it is doable, and if you specify funds, it will actually do a successful balance transfer, but really doesn't do anything else. In that case, we should keep the wording the same, but maybe clarify these details somewhere? (or is that beyond the reference docs?)

Reply:
> it will actually do a successful balance transfer, but really doesn't do anything else.

Yeah! 

> but maybe clarify these details somewhere? (or is that beyond the reference docs?)

Tbh, If I were to come up with this question I would have ask you.

Reply:
I can make this clearer inline with the call() function: 

/// Makes a call to an account, optionally transferring some balance.
/// If the account is a smart-contract account, the associated code will be executed and any fees will be transferred.
/// If the account is a regular account, any fees will be transferred.
/// If no account exists, a regular account will be created and any fees will be transferred. _[I need to double check this]_

Reply:
The last one seems good to me. Although there is a subtlety that if the call's value is below `existential_deposit` (aka `minimum_balance`) then the contract can not be created and the whole `call`/`instantiate` fails with an error.

Reply:
I think you can just ignore the last line, since this concept is already part of a balance transfer (account creation logic)

Reply:
thought about excluding it as well, atm I do want to include this for completeness and to signal that smart-contracts can call non-existent accounts